core/domain/fs_domain.py

```diff
@@ -325,8 +325,9 @@ def commit(self, filepath, raw_bytes, mimetype=None):
             raw_bytes: str. The content to be stored in the file.
             mimetype: str. The content-type of the file.
         """
-        if mimetype != 'application/octet-stream':
-            raw_bytes = python_utils.convert_to_bytes(raw_bytes)
+        raw_bytes = (
```

Comment:
> The current logic won't require any changes if we move from py2 to py3. This one will work because protobuf SerializeToString() method directly yields bytes data (which takes care of py2 vs py3) and hence we can just go ahead and store it without requiring convert_to_bytes. The convert_to_bytes is required to store textual data when file is opened in *b (binary) mode. In that case, we can't store text directly and need to bytes conversion but here since we already have bytes we don't need to do it again. So this code works on both py2 and py3.

Ah ok -- these are great details to highlight. Could you please update the docstring and add comments appropriately? Don't make them proto-specific -- basically, have the Args say that if a mimetype of XXX is used, the raw_bytes should be of type YYY, otherwise they should be of type ZZZ -- and that callers must adhere to this contract.

Reply:
Done